universal-application-tool-0.0.1/app/services/applicant/question/DateQuestion.java

```diff
@@ -0,0 +1,122 @@
+package services.applicant.question;
+
+import com.google.common.collect.ImmutableSet;
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import services.Path;
+import services.applicant.ValidationErrorMessage;
+import services.question.types.DateQuestionDefinition;
+import services.question.types.QuestionType;
+
+public class DateQuestion implements PresentsErrors {
+
+  private final ApplicantQuestion applicantQuestion;
+  private Optional<Long> dayValue;
+  private Optional<Long> monthValue;
+  private Optional<Long> yearValue;
+
+  public DateQuestion(ApplicantQuestion applicantQuestion) {
+    this.applicantQuestion = applicantQuestion;
+    assertQuestionType();
+  }
+
+  @Override
+  public boolean hasQuestionErrors() {
+    return !getQuestionErrors().isEmpty();
+  }
+
+  @Override
+  public ImmutableSet<ValidationErrorMessage> getQuestionErrors() {
+    if (!isAnswered()) {
+      return ImmutableSet.of();
+    }
+    return ImmutableSet.of();
+  }
+
+  @Override
+  public boolean hasTypeSpecificErrors() {
+    return !getAllTypeSpecificErrors().isEmpty();
+  }
+
+  @Override
+  public ImmutableSet<ValidationErrorMessage> getAllTypeSpecificErrors() {
+    // There are no inherent requirements in a date question.
```

Comment:
question (non-blocking): Do we need to check anything like  day must be between 1-31; month must be between 1-12, etc?

Reply:
+1. I think we'll want to do some date-specific validation. I think it's fine to leave this as TODO for now, but the comment that's currently here should be removed.